starknet/src/space/space.cairo

```diff
@@ -5,28 +5,71 @@ use sx::types::{
 
 #[starknet::interface]
 trait ISpace<TContractState> {
-    // State 
+    // -- View functions --
+
+    /// Returns the owner of the contract.
     fn owner(self: @TContractState) -> ContractAddress;
-    fn max_voting_duration(self: @TContractState) -> u32;
+
+    /// The voting delay to wait for between a proposal is created and the start of the voting period.
+    fn voting_delay(self: @TContractState) -> u32;
+
+    /// Returns the minimum voting duration. Once this duration has elapsed,
+    /// a proposal can be finalized. Most DAOs should set `min_voting_duration` and
+    /// `max_voting_duration` to the same value.
     fn min_voting_duration(self: @TContractState) -> u32;
+
+    /// Returns the maximum voting duration. Once this duration has elasped, a proposal
+    /// can be finalized. No other votes can come in.
+    fn max_voting_duration(self: @TContractState) -> u32;
+
+    /// Returns the DAO URI. See https://ethereum-magicians.org/t/erc-4824-decentralized-autonomous-organizations/8362
     fn dao_uri(self: @TContractState) -> Array<felt252>;
+
+    /// The next proposal id. Starts at 1.
     fn next_proposal_id(self: @TContractState) -> u256;
-    fn voting_delay(self: @TContractState) -> u32;
+
+    /// The allow list of authenticators. Only allowed authenticators can interact with the `propose`,
+    /// `udpate_proposal` and `vote` functions.
     fn authenticators(self: @TContractState, account: ContractAddress) -> bool;
+
+    /// An array of allowed voting strategies. Voting strategies are used to determine the voting power of a voter.
     fn voting_strategies(self: @TContractState, index: u8) -> Strategy;
+
+    /// The current active voting strategies. Each bit represents whether the proposal is active or not.
+    /// These are stored at the proposal level when a proposal is created, meaning when a new voting strategy is added or removed
+    /// it only affects subsequent proposals, not ongoing ones.
     fn active_voting_strategies(self: @TContractState) -> u256;
+
+    /// The next voting strategy index. Starts at 0.
```

Comment:
The index of the next available voting strategy. Starts at 0

Reply:
6f348d79f7d366db2741e52a881e17708515063c

Reply:
97aa327876b9eaa8c501dbb0b9381b6b813065f8